core/pkg/monitor/openmetrics.go

```diff
@@ -0,0 +1,279 @@
+package monitor
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-retryablehttp"
+	"github.com/wandb/wandb/core/internal/clients"
+	"github.com/wandb/wandb/core/pkg/observability"
+	spb "github.com/wandb/wandb/core/pkg/service_go_proto"
+
+	"github.com/prometheus/common/expfmt"
+)
+
+// Filter represents a filter to apply to metrics collected from an OpenMetrics endpoint.
+//
+// Provides a fine-grained control over which metrics to collect.
+// OpenMetrics filters can be specified via wandb.Settings in one of the two formats:
+//   - {"metric regex pattern, including endpoint name as prefix": {"label": "label value regex pattern"}}
+//   - ("metric regex pattern 1", "metric regex pattern 2", ...)
+//
```

Comment:
it was hard for me to understand this, maybe a specific example would be helpful (or adding this documentation in the proto part- didn't check if it is there)